voxblox_global_planner/src/skeleton_global_planner.cpp

```diff
@@ -221,4 +243,26 @@ void SkeletonGlobalPlanner::skeletonize(voxblox::Layer<voxblox::EsdfVoxel> *esdf
   skeleton_generator_.setMinGvdDistance(constraints_.robot_radius);
 }
 
+bool SkeletonGlobalPlanner::getNearestFreeSpaceToPoint(const Eigen::Vector3d& pos, 
+    Eigen::Vector3d& new_pos) const {
+  Eigen::Vector3d final_pos = pos;
+  double distance = 0.0;
+  Eigen::Vector3d gradient = Eigen::Vector3d::Zero();
+
+  const size_t max_iterations = 100;
```

Comment:
check max_iteration to always return a location.

Reply:
i have set it to check upto 10 * robot_radius. should we really do search till not found? even with 10*robot_radius, it sometimes crashes.